tests/checkout/tree.c

```diff
@@ -973,9 +973,86 @@ void test_checkout_tree__filemode_preserved_in_index(void)
 	git_commit_free(commit);
 
 
+	/* Finally, check out the text file again and check that the exec bit is cleared */
+	cl_git_pass(git_oid_fromstr(&executable_oid, "cf80f8de9f1185bf3a05f993f6121880dd0cfbc9"));
+	cl_git_pass(git_commit_lookup(&commit, g_repo, &executable_oid));
+
+	cl_git_pass(git_checkout_tree(g_repo, (const git_object *)commit, &opts));
+	cl_assert(entry = git_index_get_bypath(index, "a/b.txt", 0));
+	cl_assert_equal_i(0100644, entry->mode);
+
+	git_commit_free(commit);
+
+
 	git_index_free(index);
 }
 
+mode_t read_filemode(const char *path)
+{
+	git_buf fullpath = GIT_BUF_INIT;
+	struct stat st;
+	mode_t result;
+
+	git_buf_joinpath(&fullpath, "testrepo", path);
+	cl_must_pass(p_stat(fullpath.ptr, &st));
+
+	result = GIT_PERMS_IS_EXEC(st.st_mode) ? 0100755 : 0100644;
```

Comment:
We do have defines for this to avoid magic numbers, even if they're rather recognizable.